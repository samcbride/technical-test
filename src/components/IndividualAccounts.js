import React from "react";
import PropTypes from "prop-types";

const IndividualAccounts = (props) => {
  const { sports, kids, music, news, movies } = props;
  return (
    <div className="individual-channels" data-testid="individual-channels">
      <div className="individual-channels__sports">{sports}</div>
      <div className="individual-channels__kids">{kids}</div>
      <div className="individual-channels__music">{music}</div>
      <div className="individual-channels__news">{news}</div>
      <div className="individual-channels__movies">{movies}</div>
    </div>
  );
};

export default IndividualAccounts;

IndividualAccounts.propTypes = {
  sports: PropTypes.string,
  kids: PropTypes.string,
  music: PropTypes.string,
  news: PropTypes.string,
  movies: PropTypes.string,
};
