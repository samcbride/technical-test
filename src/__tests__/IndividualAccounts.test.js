import React from "react";
import IndividualAccounts from "../components/IndividualAccounts";
import { render } from "@testing-library/react";

describe("IndividualAccounts", () => {
  const validProps = {
    sports: "Sports",
    kids: "Kids",
    music: "Music",
    news: "News",
    movies: "Movies",
  };
  it("renders correctly", () => {
    const { asFragment } = render(
      <IndividualAccounts
        sports={validProps.sports}
        kids={validProps.kids}
        music={validProps.music}
        news={validProps.news}
        movies={validProps.movies}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
  it("renders correct values for props", () => {
    const { getByText } = render(
      <IndividualAccounts
        sports={validProps.sports}
        kids={validProps.kids}
        music={validProps.music}
        news={validProps.news}
        movies={validProps.movies}
      />
    );
    expect(getByText("Sports")).toHaveClass("individual-channels__sports");
    expect(getByText("Kids")).toHaveClass("individual-channels__kids");
    expect(getByText("Music")).toHaveClass("individual-channels__music");
    expect(getByText("News")).toHaveClass("individual-channels__news");
    expect(getByText("Movies")).toHaveClass("individual-channels__movies");
  });
});
