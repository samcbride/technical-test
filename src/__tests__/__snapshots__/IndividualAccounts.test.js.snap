// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndividualAccounts renders correctly 1`] = `
<DocumentFragment>
  <div
    class="individual-channels"
    data-testid="individual-channels"
  >
    <div
      class="individual-channels__sports"
    >
      Sports
    </div>
    <div
      class="individual-channels__kids"
    >
      Kids
    </div>
    <div
      class="individual-channels__music"
    >
      Music
    </div>
    <div
      class="individual-channels__news"
    >
      News
    </div>
    <div
      class="individual-channels__movies"
    >
      Movies
    </div>
  </div>
</DocumentFragment>
`;
